packages/taco/test/conditions/context.test.ts

```diff
@@ -303,7 +276,7 @@ describe('context', () => {
       }
     }
 
-    async function makeAuthSignature(userAddressParam: string) {
+    async function makeAuthSignature(userAddressParam: string, scheme: string) {
```

Comment:
Does it make sense to simplify this?

```suggestion
    async function makeAuthSignature(scheme: string) {
      const userAddressParam = scheme == "SSO4361" ? USER_ADDRESS_PARAM_EXTERNAL_EIP4361 : USER_ADDRESS_PARAM_DEFAULT

```

Reply:
i'm not sure on this suggestion - it feels like it could get messy when we add new auth types

Reply:
Maybe you could expand on "messy" a bit more to help me understand?

I may be missing something, but, at least currently there is one-to-one relationship between scheme and userAddressParam i.e. if we know the scheme we know the relevant user address param for the signature. I question using `userAddressParam` as a parameter because having to provide both is redundant. So the question may be, is there a world where we continue to pass only a scheme, but don't know the userAddressParam to use? 

If you are thinking about other types of `AuthSignature` (non-user address param), then perhaps the helper method should be changed to `makeUserAddressAuthSignature(scheme: string)`.

Reply:
i think this relates to your comment below - if the scheme is `SSO4361`, the `userAddressParam` could be `USER_ADDRESS_PARAM_EXTERNAL_EIP4361` or `USER_ADDRESS_PARAM_DEFAULT`

Reply:
Actually, based on my subsequent suggestion here - https://github.com/derekpierre/taco-web/pull/1#discussion_r1950892386, it now makes sense to keep userAddressParam. Ignore the above.

Reply:
> i think this relates to your comment below - if the scheme is `SSO4361`, the `userAddressParam` could be `USER_ADDRESS_PARAM_EXTERNAL_EIP4361` or `USER_ADDRESS_PARAM_DEFAULT`
> 
> I'd just prefer not to maintain a mapping, and instead allow different combos to be passed in

Haha jinx. Indeed - totally right.